cmd/publishing-bot/main.go

```diff
@@ -119,10 +119,14 @@ func main() {
 		glog.Fatalf("No rules file provided")
 	}
 
+	runChan := make(chan bool, 1)
+	defer close(runChan)
```

Comment:
wouldn't close it. might lead to a panic from the server on shutdown.

Reply:
Properly handle SIGINT and SIGTERM if you want graceful termination.

Reply:
we probably don't care about graceful termination, but would be interesting what happen to a running bot when something kill the pod in a middle of the sync.

Reply:
Yeah, I think you do care. We had issues in prow because we werent gracefully terminating and bots would be killed in the middle of an update, leaving partial Github updates behind.